### DISCUSSION TITLE
How to install in home assistant

### DISCUSSION THREAD

**Discussion by eamonnogorman** [cereal2nd/velbus-aio#81] [General]
Hi,

Sorry for probably a fact question, but how do I replace my existing velbus integration in home assistant with this one ?

Thanks,
Eamon

**Comment by cereal2nd** [comment]
Since the 2022.5 release this is the one bundled with home-adsistant

**Reply by eamonnogorman** [reply]
Thanks, it would appear my vmb1dm is not working so as it is detected but can't operate it, do we know if this is supported or is there a way of adding support for it ?

**Reply by cereal2nd** [reply]
Do you get any error message in the home-assistant log?

**Reply by eamonnogorman** [reply]
Thanks, the home assistant logs say

 "none type" object is not callable

Regards
Eamon

**Comment by cereal2nd** [comment]
can you get me a bit more info from the logs?
20 lines before and 20 lines after this error?

**Reply by eamonnogorman** [reply]
Hi,

Thanks again, hope the logs below are enough, i think I captured a few events!


[velbus HA LOGS.txt](https://github.com/Cereal2nd/velbus-aio/files/12260635/velbus.HA.LOGS.txt)

**Reply by eamonnogorman** [reply]
Hi cereal2nd, were those logs any good or do you need them in a different format?

**Comment by eamonnogorman** [comment]
Hi cereal2nd,  were those logs any good or do you need them in a different format?

**Comment by sidlgor** [comment]
Hi, I have a fork (sidlgor) of the code that I would like to load in home assistant but I cannot figure out on how to do it.
I already copied the implementation tree under custom_components but the integration is not loaded and there is also nothing in the logs.
Could you please give me a hint on how to load a new version of the code from source in home assistant (the code is already tested with examples code independent of home assistant) 
Any help appreciated

**Reply by niobos** [reply]
Yeah, it's a pain to set up...
The official docs are [here](https://developers.home-assistant.io/docs/creating_integration_manifest/#custom-requirements-during-development--testing), but I always struggle to get it to work in my docker containers.

**Reply by sidlgor** [reply]
I am also running HAS docker, so far so good I hope.
I have the impression I completely misunderstand the installation procedure. I just copied the installation tree [https://github.com/sidlgor/velbus-aio/tree/master/velbusaio](url) to the custom_components directory which might be very naive. There seems to be not even a manifest.json file in there. In fact I don't find a manifest.json file at all.
I installed several custom integrations (luxtronix, huawei-solar, frigate, ...) from HACS  without (to many) problems, but installation from a development project (py sources) seems to be completely different. Please could you guide me the right direction ? Do I have to use custom_components at all? If affirmative, do you have an example of the correct file structure, content? 
Would be a great help!

**Reply by niobos** [reply]
I copied the [Velbus integration from the official Home Assastant repo](https://github.com/home-assistant/core/tree/dev/homeassistant/components/velbus/) into `custom_components`. But it needed some fiddling with the `manifest.json` to get that to work. From memory, I think I needed to add a `version` attribute.

Next, you can change the dependencies to point to your own velbusaio fork. That part is explained in [the docs](https://developers.home-assistant.io/docs/creating_integration_manifest/#custom-requirements-during-development--testing).

But I didn't manage to get a local velbusaio directory working, and HA tries to be helpful to not download the same dependency twice, so I ended up bumping the version-number and pushing every time I changed something. I'm trying to find an easier way, but haven't found it yet.

**Comment by sidlgor** [comment]
Thanks a lot for help, but still strugling ...

 >>> I copied the Velbus integration from the official Home Assastant 
repo 
<https://github.com/home-assistant/core/tree/dev/homeassistant/components/velbus/> 
into custom_components. But it needed some fiddling with the 
manifest.json to get that to work. From memory, I think I needed to add 
a version attribute.
Great to have this reference, looks quit essential to understand


 >>>Next, you can change the dependencies to point to your own velbusaio 
fork. That part is explained in the docs 
<https://developers.home-assistant.io/docs/creating_integration_manifest/#custom-requirements-during-development--testing>.

Unsure about this: did you mean "requirements" instead of "dependencies" 
? (dependencies are set to usb)

What would be the correct requirement for my fork ? What should <git 
ref> look like?

  "requirements": ["velbus-aio @ 
git+https://github.com/sidlgor/velbus-aio.git@<git ref>"]

> >>> But I didn't manage to get a local velbusaio directory working, and 
>HA tries to be helpful to not download the same dependency twice, so I 
>ended up bumping the version-number and pushing every time I changed 
>something. I'm trying to find an easier way, but haven't found it yet.
>
Really don't mind about that at this point 🤪 I would be very happy to 
get anything working!

**Comment by sidlgor** [comment]
One more thought: instead of adding a requirement to a git fork, 
wouldn't it work to just copy (merge) the files from the fork into the 
custom_components? At that moment we could have velbusaio completely 
local?


------ Original Message ------
From "niobos" ***@***.***>
To "Cereal2nd/velbus-aio" ***@***.***>
Cc "sidlgor" ***@***.***>; "Comment" 
***@***.***>
Date 24/06/2024 12:22:20
Subject Re: [Cereal2nd/velbus-aio] How to install in home assistant 
(Discussion #81)
